### ISSUE TITLE
Support selecting multiple files at once

### ISSUE THREAD

**Issue by jesseduffield** [fyne-io/fyne#1082]
### Is your feature request related to a problem? Please describe:
I want to be able to select multiple files in the popup file picker.

### Is it possible to construct a solution with the existing API?
I'm not too sure, I haven't looked into the api

### Describe the solution you'd like to see:
In javascript the file picker api always returns an array of files (or perhaps filenames, I can't recall). The typical situation will be when one file is chosen meaning the array contains only a single element, but it's always returning an array. I'd like to see the same happening here. As for how to make it work I've got no idea but happy to fork the repo and try my hand at it if somebody can point me in the right direction.

Worth mentioning my use case only involves desktop OS's, so if this is only an issue to implement on mobile, I'm happy to use a flag to say 'this is only a non-mobile app', but I'm assuming the ethos of this repo is to be very cross-compatible.

**Comment by andydotxyz** [comment]
Thanks, this indeed does need to be added - not yet sure how well this will be supported on mobile - though there already some desktop specific APIs (in driver/desktop). 

The dialog code is in the ‘dialog’ package if you want to look (the mobile specifics are in the fyne-io/mobile repo).

**Comment by muktihari** [comment]
Hi folks! I just getting started with fyne to build a GUI app targeting desktop only and I get stuck on this as I want to select multiple files at once. Is there any update regarding this feature? Or maybe if there is any workaround to do it for desktop, can you point me to the articles or documentation. Thank you!

**Comment by andydotxyz** [comment]
You could have your app load the file open dialog multiple times to add multiple files. Or use the folder open.

**Comment by muktihari** [comment]
Thanks for the suggestions. 

Opening file dialog multiple times does not seem like a viable option since there are many files to select. And for using folder open, I don't have any idea how to implement it in a way to still deliver a good UX for users, it's like either users need to select the files after selecting folder (twice selection) or ask user to move their files to a certain folder before proceeds (a bit hazzle). 

I guess I'll wait the update. Thanks.

**Comment by andydotxyz** [comment]
> it's like either users need to select the files after selecting folder (twice selection) or ask user to move their files to a certain folder before proceeds (a bit hazzle).

Oh, they won't be using a directory full of files? Sorry it was just an idea in case.

**Comment by muktihari** [comment]
> Oh, they won't be using a directory full of files? 

Yes, only selected files. What I'm trying to do is, open multiple files at once, aggregate data from those files with multiple options to select, then save the aggregated data into a new file. The target users are non-tech people so I think it's good to have a GUI app. 

> Sorry it was just an idea in case.

No worries :beers:

**Comment by andydotxyz** [comment]
Makes sense.

One last thought - open a directory then display a panel listing all the files with checkboxes.
Your code could then iterate each selected opening readers and processing.
Depending on your user it may actually be easier because I think multiple-select in file dialogs is typically a modifier-key-hold scenario.

**Comment by silasrm** [comment]
None solution?

**Comment by andydotxyz** [comment]
When there are updates they will be posted here. 
Folk generally work on features that are a priority for them - if you want to see a speedier resolution you can always consider sponsoring.

**Comment by silasrm** [comment]
I understand, @andydotxyz . I solved this problem using github.com/ncruces/zenity.

```go
files, err := zenity.SelectFileMultiple(
			zenity.Filename(``),
			zenity.FileFilters{
				{"ZIP files", []string{"*.zip"}, false},
			})

		if err != nil {
			return
		}

		log.Println("Files: ", files)
```
